sal.js

```diff
@@ -0,0 +1,108 @@
+// minisearch accesslayer
+import { crocks, MiniSearch } from "./deps.js";
+
+const { tryCatch, compose, constant, identity  } = crocks;
+
+const foldWith = fn => m =>
+  m.either(
+    (e) => Promise.reject({ ok: false, message: "SAL:" + e.message }),
+    (_) => Promise.resolve(fn(_)),
+  )
+const fold = foldWith(constant({ok: true}))
```

Comment:
don't think this is formatted with `deno lint`

Reply:
Ugh